packages/xfsprogs/0001-mkfs-source-defaults-from-config-file.patch

```diff
@@ -0,0 +1,46 @@
+From 46605e1f1d67bf2344ffab9447e763ebc13f6387 Mon Sep 17 00:00:00 2001
+From: Sparks Song <shijiao@amazon.com>
+Date: Thu, 14 Nov 2024 00:17:10 +0000
+Subject: [PATCH] mkfs: source defaults from config file
+
+---
+ mkfs/xfs_mkfs.c | 16 ++++++++++++++++
+ 1 file changed, 16 insertions(+)
+
+diff --git a/mkfs/xfs_mkfs.c b/mkfs/xfs_mkfs.c
+index 6d2469c..bc76b80 100644
+--- a/mkfs/xfs_mkfs.c
++++ b/mkfs/xfs_mkfs.c
+@@ -14,6 +14,7 @@
+ #include "libfrog/dahashselftest.h"
+ #include "proto.h"
+ #include <ini.h>
++#include <unistd.h>
+ 
+ #define TERABYTES(count, blog)	((uint64_t)(count) << (40 - (blog)))
+ #define GIGABYTES(count, blog)	((uint64_t)(count) << (30 - (blog)))
+@@ -4369,6 +4370,21 @@ main(
+ 	 * printf(_("Default configuration sourced from %s\n"), dft.source);
+ 	 */
+ 
++	char *default_config_path = "/usr/share/xfs/mkfs.xfs.conf";
```

Comment:
The more idiomatic way to do this would be to use `#define` to set a constant like this. It helps signal to the reader this never changes.